src/datepicker/datepicker-input.ts

```diff
@@ -288,6 +290,7 @@ export class NgbInputDatepicker implements OnChanges,
       config: NgbInputDatepickerConfig) {
     ['autoClose', 'container', 'positionTarget', 'placement'].forEach(input => this[input] = config[input]);
     this._zoneSubscription = _ngZone.onStable.subscribe(() => this._positioning.update());
+    this.closed.subscribe(this._destroyCloseHandlers$);
```

Comment:
Could you do the same as in dropdown and just `this._destrotCloseHandlers$.next()` inside the `close()` method where we actually firing `this.close.next()` to avoid having this subscription?

Also like this we never unsubscribe

Reply:
Fair enough , there's no other reason to call this.close.next anyway.